app/controllers/featured_circuits_controller.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+class FeaturedCircuitsController < ApplicationController
+  before_action :authenticate_user!, only: [:feature]
+
+  def index
+    @projects = FeaturedCircuit.includes(:project).map(&:project)
```

Comment:
`includes` is still expensive in terms of memory. Can you try
```ruby
class Project
  has_one :featured_circuit
end
```

```suggestion
    @projects = Project.joins(:featured_circuit)
```

Reply:
Done, `joins` also saved us a query as compared to `includes`.